libsolidity/analysis/TypeChecker.cpp

```diff
@@ -1890,6 +1890,27 @@ void TypeChecker::endVisit(BinaryOperation const& _operation)
 	_operation.annotation().isLValue = false;
 	_operation.annotation().isConstant = false;
 
+	if (_operation.getOperator() == Token::Equal || _operation.getOperator() == Token::NotEqual)
```

Comment:
Inequalities are disallowed, right? Would be good to have an assert against that here. If, for some weird reason they are not, they have the same problem as equality.

Reply:
They're not disallowed.

Reply:
Ah, never mind - I thought you were referring to `!=`, and not `<, >` - these are in fact disallowed, and are already covered by tests.